### PR TITLE
Fix Contains method for inverted rects

### PR DESCRIPTION
**Pull request by SprinterDave**
Fix for https://github.com/ScottPlot/ScottPlot/issues/4596

### CODE REVIEW

File: src/ScottPlot5/ScottPlot5 Tests/Unit Tests/UnitTests/PrimitiveTests/CoordinateRectTests.cs [hunk 1]
```diff
@@ -38,4 +38,20 @@ public void Test_CoordinateRect_Expanded()
         CoordinateRect cRect2 = cRect.Expanded(coord);
         cRect2.Contains(coord).Should().BeTrue();
     }
+
+    [TestCase(0, 0, 1, 1, 0.5, 0.5, true, TestName = "Test Contains with no inverted")]
+    [TestCase(0, 0, 1, -1, 0.5, -0.5, true, TestName = "Test Contains with inverted X")]
+    [TestCase(0, 0, -1, 1, -0.5, 0.5, true, TestName = "Test Contains with inverted Y")]
+    [TestCase(0, 0, -1, -1, -0.5, -0.5, true, TestName = "Test Contains with inverted X and Y")]
```
Comment (bclehmann) [inline src/ScottPlot5/ScottPlot5 Tests/Unit Tests/UnitTests/PrimitiveTests/CoordinateRectTests.cs:45]: These test cases are a little hard to parse, just because there's 6 numbers in there, I'd prefer to pass them in as a struct or something, just so that it's more clear to read.

I might suggest something like this:
```cs
private static readonly (CoordinateRect rect, Coordinates pt)[] ContainsTestCases = {
    (new(0, 1, 0, 1), new(0.5, 0.5)), // Uninverted Rect
    (new(0, 1, 0, -1), new(0.5, -0.5)), // inverted X
    (new(0, -1, 0, 1), new(-0.5, 0.5)), // inverted Y
    (new(0, -1, 0, -1), new(-0.5, -0.5)) // inverted X and Y
};

[Test, TestCaseSource(nameof(ContainsTestCases))]
public void Test_CoordinateRect_Contains((CoordinateRect rect, Coordinates pt) testCase)
{
    Assert.That(testCase.rect.Contains(testCase.pt), Is.True);
}
```

Note that it's equivalent, the order of the numbers is just different because the `CordinateRect` constructor takes `left, right, bot, top` and your code lists them in the order `left, bot, right, top`

**Review by bclehmann — APPROVED** [review]
Looks good, thanks for including tests.

I just had one comment to make the tests more readable, I'd rather pass the test params in as structs just because my eyes start to glaze over trying to keep 6 params in the right order.

File: src/ScottPlot5/ScottPlot5 Tests/Unit Tests/UnitTests/PrimitiveTests/CoordinateRectTests.cs (hunk 1 above)
Comment (swharden) [reply src/ScottPlot5/ScottPlot5 Tests/Unit Tests/UnitTests/PrimitiveTests/CoordinateRectTests.cs:45]: Thanks for this feedback Ben! While I do agree, I'm okay merging this as it is given the situation... I'm grateful when anonymous internet contributions with tests at all 🙃

**Comment by swharden** [comment]
Thanks @SprinterDave! BTW I was really inspired when I participated in FIRST as a teen in the early 2000s and it's great to see others mentoring in that program today. 

Hope your 2025 is off to a great start,
Scott
